terraform/iam/iam-general.txt

```diff
@@ -0,0 +1,79 @@
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "CFGated",
+            "Effect": "Allow",
+            "Action": [
+                "*"
+            ],
+            "Resource": [
+                "arn:aws:cloudformation:<YOUR-REGION>:<IAM-ACCOUNT-NUM>:stack/<YOUR-STACK-NAME>/*",
+                "arn:aws:cloudformation:<YOUR-REGION>:<IAM-ACCOUNT-NUM>:stackset/<YOUR-STACK-NAME>:*"
+            ]
+        },
+        {
+            "Sid": "S3Gated",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": [
+                "arn:aws:s3:::<YOUR-STACK-NAME>-cf-assets/*",
+                "arn:aws:s3:::<YOUR-STACK-NAME>-cf-assets"
+            ]
+        },
+        {
+            "Sid": "IAMPassRoleGated",
+            "Effect": "Allow",
+            "Action": [
+                "iam:PassRole"
+            ],
+            "Resource": [
+                "arn:aws:iam::<IAM-ACCOUNT-NUM>:role/<YOUR-STACK-NAME>"
+            ]
+        },
+        {
+            "Sid": "STSGated",
+            "Effect": "Allow",
+            "Action": [
+                "sts:AssumeRole"
+            ],
+            "Resource": [
+                "arn:aws:iam::<IAM-ACCOUNT-NUM>:role/<YOUR-STACK-NAME>*"
+            ]
+        },
+        {
+            "Sid": "ungated",
+            "Effect": "Allow",
+            "Action": [
+                "cloudformation:ActivateType",
+                "cloudformation:BatchDescribeTypeConfigurations",
+                "cloudformation:CreateStackSet",
+                "cloudformation:CreateUploadBucket",
+                "cloudformation:DeactivateType",
+                "cloudformation:DeregisterType",
+                "cloudformation:Describe*",
+                "cloudformation:EstimateTemplateCost",
+                "cloudformation:List*",
+                "cloudformation:PublishType",
+                "cloudformation:RegisterPublisher",
+                "cloudformation:RegisterType",
+                "cloudformation:SetTypeConfiguration",
+                "cloudformation:SetTypeDefaultVersion",
+                "cloudformation:TestType",
+                "cloudformation:ValidateTemplate",
+                "ec2:DescribeAvailabilityZones",
+                "eks:Describe*",
+                "eks:List*",
+                "iam:Get*",
+                "iam:List*",
+                "kms:DescribeKey",
+                "s3:CreateJob",
+                "s3:Get*",
+                "s3:List*",
+                "s3:PutAccountPublicAccessBlock",
+                "s3:PutStorageLensConfiguration"
+            ],
+            "Resource": "*"
+        }
```

Comment:
need another section like this to clean up vaults and recovery points 
```
        {
            "Sid": "BackupGated",
            "Effect": "Allow",
            "Action": "backup:*",
            "Resource": [
                "arn:aws:backup:eu-central-1:415280201047:backup-vault:domino-dazn*",
                "arn:aws:backup:eu-central-1:415280201047:recovery-point*"
            ]
        },
```

Reply:
couldn't find any way to gate the recovery point piece to just our resources though, so might be possible to improve upon that